packages/mobile/src/verify/saga.ts

```diff
@@ -197,19 +204,24 @@ function* getCodeForIssuer(issuer: string): Generator<any, AttestationCode | und
 }
 
 // Get the code from the store if it's already there, otherwise wait for it
-function* waitForAttestationCode(issuer: string): Generator<any, AttestationCode, any> {
+function* waitForAttestationCode(issuer: string): Generator<any, AttestationCode | null, any> {
   Logger.debug(TAG + '@waitForAttestationCode', `Waiting for code for issuer ${issuer}`)
   const code = yield call(getCodeForIssuer, issuer)
   if (code) {
     return code
   }
 
-  while (true) {
-    const action: ReturnType<typeof inputAttestationCode> = yield take(inputAttestationCode.type)
-    if (action.payload.issuer === issuer) {
-      return action.payload
-    }
-  }
+  const {
```

Comment:
Not canceling here meant that when a user re-requested codes there were two listeners per code, the original one and the one created for the resend, so when the user put in the code for the issuer 2 codes were processed and one failed, causing the user to fail out of the flow.